nemo/lightning/pytorch/strategies/megatron_strategy.py

```diff
@@ -715,8 +715,10 @@ def load_model_state_dict(self, checkpoint: Mapping[str, Any], strict: bool = Tr
         assert self.megatron_parallel is not None
 
         _strategy_lib.load_model_state_dict(self.megatron_parallel, checkpoint, strict=strict)
-        for opt in self.optimizers:
-            opt.reload_model_params()
+
+        if not 'optimizer' in checkpoint:
```

Comment:
Should this also check for `optimizer_states` key?

Reply:
I assumed it would be empty https://github.com/NVIDIA/NeMo/blob/main/nemo/lightning/pytorch/strategies/megatron_strategy.py#L644

Reply:
Makes sense 👍